### PR TITLE
Support responding different content types based on the accept header and different response codes.

### PR DESCRIPTION
**Pull request by VinulaUthsara**
Inline prototyping has been improved to read accept headers and response codes, allowing users to define multiple payloads under a response code per resource. Users are able to specify which response code and content-type they require and will receive accordingly. 

Response codes without pre-defined payloads will be shown as commented out string, allowing users to modify later if needed.

Screenshot from 2020-06-24 08-19-50

Screenshot from 2020-06-24 08-20-01

Fixes : https://github.com/wso2/product-apim/issues/7782

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java [hunk 1]
```diff
@@ -219,78 +205,99 @@ private String getSchemaExample(Model model, Map<String, Model> definitions, Has
...
-                responseSectionString +
-                " if (responseCode == null) {\n\n" +
-                "  mc.setProperty('HTTP_SC', \"" + minResponseCode + "\");\n" +
-                "  mc.setProperty('CONTENT_TYPE', 'application/" + minResponseType + "');\n" +
-                "  mc.setPayload" + minResponseType.toUpperCase() + "(response" + minResponseCode + minResponseType + ");\n\n" +
-                "} else {\n\n" +
-                "  mc.setProperty('CONTENT_TYPE', 'application/json');\n" +
-                "  mc.setPayloadJSON(response501json);\n\n" +
-                "}";
+    private String getResponseCondtionsSection(boolean hasJsonPayload, boolean hasXmlPayload) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS2Parser.java:273]: Please fix the typo

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java [hunk 2]
```diff
@@ -257,82 +246,102 @@ private String getJsonExample(Schema model, Map<String, Schema> definitions) {
...
-     *
-     * @param responseCode response Entry Code
-     * @param type mediaType (Json/Xml)
-     * @return manualCode
-     */
-    private String getGeneratedSetResponse(String responseCode, String type) {
-        return "  mc.setProperty('HTTP_SC', \"" + responseCode + "\");\n" +
-                "  mc.setProperty('CONTENT_TYPE', 'application/" + type + "');\n" +
-                "  mc.setPayload" + type.toUpperCase() + "(response" + responseCode + type + ");";
+    private String getGenRespPayloads(String responseCode, String example, String type, boolean initialized) {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java:273]: Please use full names. getGeneratedResponsPayloads or the one that you meant to say.

**Comment by malinthaprasan** [comment]
Good Job.

Can you post a full script that was generated along with these features?

**Comment by VinulaUthsara** [comment]
**Following is a script generated for (200 Json, 200 Xml and 400 Json) payloads.**





var accept = "\""+mc.getProperty('AcceptHeader')+"\"";
var responseCode = mc.getProperty('query.param.responseCode');
var responses = [];

if (!responses[200]) {
 responses [200] = [];
}
responses[200]["application/json"] = 
{
  "account_number" : 1234,
  "account_pin" : 123,
  "account_holder" : "Vinula",
  "account_balance" : 1000,
  "phone_number" : "76866578"
}

responses[200]["application/xml"] = 
<CreateAccountDetails><account_number>987654321</account_number><account_pin>44332211</account_pin><account_holder>Shanathani</account_holder><account_balance>5000</account_balance><phone_number>9998384</phone_number></CreateAccountDetails>

if (!responses[400]) {
 responses [400] = [];
}
responses[400]["application/json"] = 
{
  "account_number" : 4321,
  "account_pin" : 321,
  "account_holder" : "aluniV",
  "account_balance" : 1,
  "phone_number" : "875668670"
}

responses[501] = [];
responses[501]["application/json"] = {
"code" : 501,
"description" : "Not Implemented"}
responses[501]["application/xml"] = <response><code>501</code><description>Not Implemented</description></response>;

if (responseCode == null) {
 responseCode = 200;
}

if (!responses[responseCode]) {
 responseCode = 501;
}

if (accept == null || !responses[responseCode][accept]) {
 accept = "application/json";
}

if (accept === "application/json") {
 mc.setProperty('CONTENT_TYPE', 'application/json');
 mc.setPayloadJSON(responses[responseCode]["application/json"]);
} else if (accept === "application/xml") {
 mc.setProperty('CONTENT_TYPE', 'application/xml');
 mc.setPayloadXML(responses[responseCode]["application/xml"]);
}

**Comment by VinulaUthsara** [comment]
**Following is the new script generated for (200 Json, 200 Xml and 400 Json) payloads with response code setting functionality.**

var accept = "\""+mc.getProperty('AcceptHeader')+"\"";
var responseCode = mc.getProperty('query.param.responseCode');
var responseCodeStr = "\""+responseCode+"\"";
var responses = [];

if (!responses[200]) {
 responses [200] = [];
}
responses[200]["application/json"] = 
{
  "account_number" : 1234,
  "account_pin" : 123,
  "account_holder" : "Vinula",
  "account_balance" : 1000,
  "phone_number" : "76866578"
}

responses[200]["application/xml"] = 
<CreateAccountDetails><account_number>987654321</account_number><account_pin>44332211</account_pin><account_holder>Shanathani</account_holder><account_balance>5000</account_balance><phone_number>9998384</phone_number></CreateAccountDetails>

if (!responses[400]) {
 responses [400] = [];
}
responses[400]["application/json"] = 
{
  "account_number" : 4321,
  "account_pin" : 321,
  "account_holder" : "aluniV",
  "account_balance" : 1,
  "phone_number" : "875668670"
}

responses[501] = [];
responses[501]["application/json"] = {
"code" : 501,
"description" : "Not Implemented"}
responses[501]["application/xml"] = <response><code>501</code><description>Not Implemented</description></response>;

if (!responses[responseCode]) {
 responseCode = 501;
}

if (responseCode == null) {
 responseCode = 200;
 responseCodeStr = "200";
}

if (accept == null || !responses[responseCode][accept]) {
 accept = "application/json";
}

if (accept === "application/json") {
 mc.setProperty('CONTENT_TYPE', 'application/json');
 mc.setProperty('HTTP_SC', responseCodeStr);
 mc.setPayloadJSON(responses[responseCode]["application/json"]);
} else if (accept === "application/xml") {
 mc.setProperty('CONTENT_TYPE', 'application/xml');
 mc.setProperty('HTTP_SC', responseCodeStr);
 mc.setPayloadXML(responses[responseCode]["application/xml"]);
}

**Pull request merged**